### PR TITLE
Add "New in v3.1" guide

### PR DESCRIPTION
**Pull request by ines**
## Description

This PR adds a "New in v3.1" usage guide, similar to the ones we did for previous minor releases. Still a work in progress, feel free to push your updates to the branch.

### Types of change
docs

## Checklist

- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: website/docs/usage/v3-1.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+| ------------------------------------------------- | -------- | -----: | -----: | ---: |
+| [`ca_core_news_sm`](/models/ca#ca_core_news_sm)   | Catalan  |        |        |      |
+| [`ca_core_news_md`](/models/ca#ca_core_news_md)   | Catalan  |        |        |      |
+| [`ca_core_news_lg`](/models/ca#ca_core_news_lg)   | Catalan  |        |        |      |
+| [`ca_core_news_trf`](/models/ca#ca_core_news_trf) | Catalan  |        |        |      |
+| [`da_core_news_trf`](/models/da#da_core_news_trf) | Danish   |        |        |      |
+
+### Resizable text classification architectures {#resizable-textcat}
+
+Previously, a trained [`TextCategorizer`](/api/textcategorizer) architectures
```
Comment (svlandeg) [inline website/docs/usage/v3-1.md:130]:
```suggestion
Previously, a trained [`TextCategorizer`](/api/textcategorizer) architecture
```

Comment (ines) [reply website/docs/usage/v3-1.md:130]: Maybe this should actually be "model"? A "trained architecture" is kinda weird. Actually, the whole sentence is weird 😂

How about:

```
Previously, the [`TextCategorizer`](/api/textcategorizer) architectures could not be resized, meaning that you couldn't add new labels to an already trained model.
```

Comment (svlandeg) [reply website/docs/usage/v3-1.md:130]: Sounds good!

File: website/docs/usage/v3-1.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+pipelines for Catalan and Danish, as well as many bug fixes and improvements.
+
+### Using predicted annotations during training {#predicted-annotations-training}
+
+By default, components are updated in isolation during training, which means
+that they don't see the predictions of any earlier components in the pipeline.
+The new
+[`[training.annotating_components]`](/usage/training#annotating-components)
+config setting lets you specify pipeline component names that should set
+annotations on the predicted docs during training. This makes it easy to use the
```
Comment (svlandeg) [inline website/docs/usage/v3-1.md:28]:
```suggestion
The new
[`[training.annotating_components]`](/usage/training#annotating-components)
config setting lets you specify pipeline components that should set
annotations on the predicted docs during training. This makes it easy to use the
```

**Pull request merged**